app/utils/wsl.js

```diff
@@ -0,0 +1,22 @@
+const path = require('path');
+
+// Valid Windows Subsystem for Linux shell executables
+const WSL_SHELL_EXECUTABLES = [
+  'bash.exe',
+  'wsl.exe',
+  'ubuntu.exe',
+  'ubuntu1804.exe',
+  'kali.exe',
+  'debian.exe',
+  'opensuse-42.exe',
+  'sles-12.exe'
+];
```

Comment:
Can you transform this into something like:
```
const WIN_SHELLL_EXECUTABLES = [
  'cmd.exe',
  'powershell.exe'
]
function isWslShell(shell) {
  return process.platform === 'win32' && !WIN_SHELL_EXECUTABLES.indexOf(shell.split(path.sep).pop()) !== -1;
}
```
Be careful with empty/missing shell config.

It would be easier to maintain like @Tyriar suggested.

Reply:
I disagree, there are many other non-WSL shells: git bash, python, gchi, etc.

Reply:
Yeah, either we have a full list for valid WSL shells, or non-WSL shells. Since the function is called `isWslShell` and the property is `isWsl`, I think it makes more sense to have a list of valid WSL shells. I'll handle the empty/missing shell configuration though.

Reply:
Oh yes, you're right.
But, in this case, it will take some effort to keep this list up to date.

Reply:
It might, it all depends on how frequent new Linux distros are going to be supported by WSL and if any of them will include different exes than the default ones. But I honestly can't think of any other way to detect if the active/running shell is WSL or not. If you have a suggestion, I'm all hears :)